test/libsolidity/semanticTests/storage/static_array_copy_cleanup.sol

```diff
@@ -0,0 +1,102 @@
+contract C {
+    struct S {
+        uint64 a;
+        uint64 b;
+        uint64 c;
+        uint64 d;
+        // All fit in one slot (4 * 64 = 256 bits)
+    }
+
+    S[5] source;
+    S[10] dest;
+    uint256 canary;
+
+    constructor() {
+        canary = type(uint256).max;
+    }
```

Comment:
```suggestion
    uint256 public canary = type(uint256).max;
```

And you can remove `getCanary()`.

Reply:
For the tests that use inline assembly to position at specific slots (i.e. `storage_boundary_*` and `long_byte_array_cleanup_*`), I'm not allowed to use storage data location with value types like `uint256`. So I will keep those with the struct canary.

Reply:
With `long_byte_array_cleanup_*` yes, you do have to use a struct. But `storage_boundary_*` use short static arrays that you position exactly at the boundary. You could use normal state vars with those just fine.